### DISCUSSION TITLE
Testing Individual Unit Tests

### DISCUSSION THREAD

**Discussion by jvanderaa** [nautobot/nautobot#1039] [Q&A] [answered]
I don't recall the exact verbiage, so I would like to defer to the exact terms by @jathanism and @glennmatthews, but it would be good to call out that you can run individual unit tests using the Invoke methods. With invoke, the individual test that you want to call is called with the name. In the Invoke tasks, that is the --label portion. This will help with your development environment.

**Comment by glennmatthews** [comment, marked as answer]
Yes, the syntax is `invoke unittest --label <label>`, for example 
- `invoke unittest --label nautobot.extras.tests` (all tests in the `nautobot.extras.tests` module)
- `invoke unittest --label nautobot.extras.tests.test_models.ConfigContextTest` (the `ConfigContextTest` class within the `nautobot.extras.tests.test_models` module) 
- `invoke unittest --label nautobot.extras.tests.test_models.ConfigContextTest.test_name_uniqueness` (specific test case within that class).

A documentation PR would be more than welcome here. :-)
